components/feature/prompts/src/test/java/mozilla/components/feature/prompts/AuthenticationDialogFragmentTest.kt

```diff
@@ -0,0 +1,182 @@
+import mozilla.components.feature.prompts.R.id
+import mozilla.components.support.test.mock
+import org.junit.Assert.assertEquals
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.mockito.Mockito.doReturn
+import org.mockito.Mockito.spy
+import org.mockito.Mockito.verify
+import org.robolectric.RobolectricTestRunner
+
+@RunWith(RobolectricTestRunner::class)
+class AuthenticationDialogFragmentTest {
+
+    private val context: Context
+        get() = ContextThemeWrapper(
+            ApplicationProvider.getApplicationContext(),
+            android.support.v7.appcompat.R.style.Theme_AppCompat
+        )
+
+    @Test
+    fun `build dialog`() {
+
+        val fragment = spy(
+            AuthenticationDialogFragment.newInstance(
+                "sessionId",
+                "title",
+                "message",
+                "username",
+                "password",
+                false
+            )
+        )
+
+        doReturn(context).`when`(fragment).requireContext()
+
+        val dialog = fragment.onCreateDialog(null)
+
+        dialog.show()
+
+        val titleTextView = dialog.findViewById<TextView>(android.support.v7.appcompat.R.id.alertTitle)
+        val messageTextView = dialog.findViewById<TextView>(android.R.id.message)
+        val usernameEditText = dialog.findViewById<EditText>(id.username)
+        val passwordEditText = dialog.findViewById<EditText>(id.password)
+
+        assertEquals(fragment.sessionId, "sessionId")
+        assertEquals(fragment.title, "title")
+        assertEquals(fragment.message, "message")
+        assertEquals(fragment.username, "username")
+        assertEquals(fragment.password, "password")
+        assertEquals(fragment.onlyShowPassword, false)
+
+        assertEquals(titleTextView.text, "title")
+        assertEquals(messageTextView.text, "message")
+        assertEquals(usernameEditText.text.toString(), "username")
+        assertEquals(passwordEditText.text.toString(), "password")
+
+        usernameEditText.setText("new_username")
+        passwordEditText.setText("new_password")
+
+        assertEquals(usernameEditText.text.toString(), "new_username")
+        assertEquals(passwordEditText.text.toString(), "new_password")
+    }
+
+    @Test
+    fun `dialog with onlyShowPassword equals true must not have a username field`() {
+
+        val fragment = spy(
+            AuthenticationDialogFragment.newInstance(
+                "sessionId",
+                "title",
+                "message",
+                "username",
+                "password",
+                true
+            )
+        )
+
+        doReturn(context).`when`(fragment).requireContext()
+
+        val dialog = fragment.onCreateDialog(null)
+
+        dialog.show()
+
+        val usernameEditText = dialog.findViewById<EditText>(id.username)
+
+        assertEquals(usernameEditText.visibility, GONE)
+    }
+
+    @Test
+    fun `dialog with not title must has default value`() {
```

Comment:
nit: dialog without a title?

Reply:
The title of the dialog can be optional and we have to provide a default value for that.
What about improving the test name to:

`when the title is not provided the dialog must has a default value`?